### PR TITLE
Basic logging functions

### PR DESCRIPTION
**Pull request by OblackatO**
- Added two new files to .gitignore
- Updated README.md
- Added a custom logger for the ChatAPI class *__get_class_logger*, with a custom anonymous formatter class *__IOFormatter*
- Added new *LOGGING_DIR* constant in const.py, and updated chatgpt-api config and logging paths. 
- Session messages are now logged to a custom file, using the already existing data keys, and look like the following: 
```
{"timestamp":"1670535995.683926", "input": "color red", "output": "Red is a color that is often associated with passion, love, and warmth. It is often seen as a bold and vibrant color that can add a lively and energetic vibe to any space. In nature, red is the color of many fruits and flowers, and is also often seen in fire and in sunsets. In art and design, red can be used to draw attention and create a sense of urgency or excitement. It is also often used to represent danger or warning.", "id": "986f79bf-414f-4850-82bd-74aa20099892", "conversation_id": "a0f8932d-07d3-483c-a5e9-6e3e0235cbaf", "parent_message_id": "986f79bf-414f-4850-82bd-74aa20099892"}
{"timestamp":"1670536023.16458", "input": "color blue", "output": "Blue is a color that is often associated with calmness and serenity. It is the color of the sky and the ocean, and is often used in design and art to create a sense of tranquility and peace. In nature, blue is the color of many flowers and plants, and is also often seen in the feathers of birds. In art and design, blue can be used to create a sense of depth and distance, and is often used in combination with other colors to create a sense of balance and harmony. Blue is also often associated with trust and reliability.", "id": "840ed997-af3f-479e-9107-4964523854c1", "conversation_id": "a0f8932d-07d3-483c-a5e9-6e3e0235cbaf", "parent_message_id": "840ed997-af3f-479e-9107-4964523854c1"}
...
```
- Fixed 3 tests, not passing in closed [merge request](https://github.com/mbroton/chatgpt-api/pull/14)

### CODE REVIEW

**Comment by OblackatO** [comment]
Never used `pre-commit` and was not able to see the flake8, mypy etc checks. Read a bit about it, installed it, and all good now. 

What do you think about [black](https://github.com/psf/black)? It is a hustle free code formatter. We could set the line length to ur standard of 79. The command would be something like: `black --line-length 79 ./chatgpt/*`

File: README.md
Comment (mbroton) [inline README.md:8]: Looks like you didn't have latest changes on your branch, because a few changes in this file are in reverse to what I've committed recently (line 8, 16, 23, 29 and 118)

File: chatgpt/api.py
Comment (mbroton) [inline chatgpt/api.py:117]: This causes `DeprecationWarning` from `httpx` package. It's recommended to use `content` argument (I also changed in 0.2.1, see: https://github.com/mbroton/chatgpt-api/pull/9/files).

File: chatgpt/api.py
Comment (mbroton) [inline chatgpt/api.py:186]: `f"{logging_path}"` - is it about converting it to a string?
If so, I think that converting it explicitly would be a better approach, having in mind Zen of Python rule (explicit is better than implicit) - `str(logging_path)`

File: chatgpt/cli.py
Comment (mbroton) [inline chatgpt/cli.py:28]: I'm wondering... if later, there's a need to pass a path to the session key file, shouldn't be `CONFIG_DIR` removed? I mean the setup flow here looks like this:
- Check if `CONFIG_DIR` exists
- Create if it doesn't exist
- Ask for a path to session key file
- Read token from given path

`CONFIG_DIR` is not really used

**Comment by mbroton** [comment]
Many changes in this PR are just in reverse to what recently had been pushed to `main` branch.
Looks like you have on your fork version <0.2.1.
I like it how you handled logging, but unfortunately I can't approve this PR before it won't be up to date with `main`.
I didn't comment any of the test files for that reason. If you want me to help you with that, just let me know.

> Never used `pre-commit` and was not able to see the flake8, mypy etc checks. Read a bit about it, installed it, and all good now.
> 
> What do you think about [black](https://github.com/psf/black)? It is a hustle free code formatter. We could set the line length to ur standard of 79. The command would be something like: `black --line-length 79 ./chatgpt/*`

I agree, pre-commit is a great tool that I started to use on a daily basis recently 😀
I use black (with line length set to 79), just haven't added it to pre-commit. I forgot 🙄

**Review by mbroton — CHANGES_REQUESTED** [review]

File: README.md
Comment (OblackatO) [reply README.md]: This should be ok now.

File: chatgpt/api.py
Comment (OblackatO) [reply chatgpt/api.py:117]: Fixed it.

File: chatgpt/api.py
Comment (OblackatO) [reply chatgpt/api.py]: f-string removed.

File: chatgpt/api.py
Comment (OblackatO) [reply chatgpt/api.py]: logging_path is now explicitly converted to str.

File: chatgpt/cli.py
Comment (OblackatO) [reply chatgpt/cli.py]: I removed the config directory... Because we need some location to store the key and logs. That would the .chatgpt-api in *home*. I also removed the confirmation to create it, because the directory would be created anyway in the logging part:
```python
if not LOGGING_DIR.exists():
    LOGGING_DIR.mkdir(parents=True, exist_ok=True)
```

There is still a warning that sates it was created.

**Comment by OblackatO** [comment]
About the tests, I would indeed need your help... I am having trouble making them work with the mock module to fake the creation of the directory. Tried with both *os.path* and *Pathlib* I removed this test *def test_cli_setup_config_file_does_not_exist_and_do_not_create(mocker):* , because confirmation is not longer ask. 

I comment out the other two:
```python
def test_cli_setup_chatgpt_dir_exist(mocker):
    """Test flow when CHATGPT_DIR exists"""


def test_cli_setup_chatgpt_dir_does_not_exist_and_create(mocker):
    """Test flow when CHATGPT_DIR does not exist"""
```

File: chatgpt/api.py
Comment (mbroton) [reply chatgpt/api.py:117]: Are you going to push it? 😄

File: key.txt
Comment (mbroton) [inline key.txt:1]: Is there a need to keep it created? If so, maybe it would be better to keep it empty, instead of keeping there something that look alike a valid key?

**Review by mbroton — COMMENTED** [review]
Please, take a look on these 2 comments I left.
After that, I'll merge it and take care of the unit tests part.

File: chatgpt/api.py
Comment (OblackatO) [reply chatgpt/api.py]: Just did it ! 🌝

File: key.txt
Comment (OblackatO) [reply key.txt]: There is no need at all to have this file. It's funny, I had added a tip on readme.md + an exception on .gitignore to avoid exactly this problem. 

I added a more general exception on .gitignore instead and removed the tip on readme.me:
- *Tip: Use a file named .session_key in chatgpt-api top directory. It will be ignored by git - see .gitignore.*

There is a test failing, but it is not related to the content of this merge request. ChatGPT is currently using CloudFare DDoS protection and the requests might have been blocked or should be done differently... Just a quick assessment, didn't look into the details.

**Pull request merged**
